umich_api/api_utils.py

```diff
@@ -30,14 +41,34 @@ def __init__(self, base_url, client_id, client_secret, api_json = None, token_ex
         :type api_json: str, optional
         :param token_expires_percent: This is a percentage of time to take off the token renewal to ensure it doesn't run out. For instance 5(%) of 3600 is 180. 
         :type token_expires_percent: int, optional
+        :param max_retries: Number of times to retry calls for all issues (timeouts, 50X status codes)
+        :type max_retries: int, optional (default is 3)
         :raises Exception: If this cannot be configured with parameters used
         :raises AttributeError: If the apis.json file is empty
         """
 
-        self.base_url = base_url
-        self.client_id = client_id
-        self.client_secret = client_secret
-        self.token_expires_percent = int(100-token_expires_percent)/100
+        self.base_url: str = base_url
+        self.client_id: str = client_id
+        self.client_secret: str = client_secret
+        self.token_expires_percent: int = int(100-token_expires_percent)/100
+
+        self.session: Session = Session()
+        self.session.headers.update({ 'accept' : 'application/json' })
+        retry = Retry(
+            total=max_retries,
+            connect=max_retries,
+            backoff_factor=0.5,
```

Comment:
`{backoff factor} * (2 ** ({number of total retries} - 1))
`
Seems like it would be fine, but wouldn't get very high with just 3 retries.